src/iface/ethernet.rs

```diff
@@ -514,16 +695,42 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         self.ip_addrs.iter().any(|probe| probe.address() == addr)
     }
 
+    /// Get the first IPv4 address of the interface.
+    #[cfg(feature = "proto-ipv4")]
+    pub fn ipv4_address(&self) -> Option<Ipv4Address> {
+        self.ip_addrs.iter()
+            .filter_map(
+                |addr| match addr {
+                    &IpCidr::Ipv4(cidr) => Some(cidr.address()),
+                    _ => None,
+                })
+            .next()
+    }
+
+    /// Check whether the interface listens to given destination multicast IP address.
+    pub fn has_mcast_group<T: Into<IpAddress>>(&self, addr: T) -> bool {
```

Comment:
This should be conditional on `feature = "proto-igmp"`, like the rest.

Reply:
I also think this function should do an early return if the address being considered is not a multicast address. This will save us the map traversal.